haystack_experimental/dataclasses/chat_message.py

```diff
@@ -0,0 +1,187 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List, Literal, Optional, Sequence, Union
+
+import filetype
+import haystack.dataclasses.chat_message
+from haystack import logging
+from haystack.dataclasses import ChatMessage as HaystackChatMessage
+from haystack.dataclasses import ChatRole, TextContent, ToolCall, ToolCallResult
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ImageContent:
+    """
+    The image content of a chat message.
+
+    :param base64_image: A base64 string representing the image.
+    :param mime_type: The mime type of the image.
+    :param detail: Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+    :param meta: Optional metadata for the image.
+    """
+
+    base64_image: str
+    mime_type: Optional[str] = None
+    detail: Optional[Literal["auto", "high", "low"]] = None
+    meta: Dict[str, Any] = field(default_factory=dict)
+
+    def __post_init__(self):
+        # mime_type is an important information, so we try to guess it if not provided
+        if not self.mime_type:
+            try:
+                # Attempt to decode the string as base64
+                decoded_image = base64.b64decode(self.base64_image)
+
+                guess = filetype.guess(decoded_image)
+                if guess:
+                    self.mime_type = guess.mime
+            except:
+                pass
```

Comment:
I think I might log a warning here if there was a decoding issue. Could indicate something wrong with the base64 encoded string right?

Reply:
Whenever we templatize `ImageContent` (e.g. in `ChatPromptBuilder`), this is expected, so using `logger.warning` might become noisy. Should we use `logger.info`?

Reply:
Ahh I see, you're saying using ImagConent with a Jinja2 template in ChatPromptBuilder will cause this to throw an exception. I agree that it'd become too noisy so probably best to leave alone for now. 

Maybe we can come back to this later if we think of a way of avoiding this when templatizing (like with your new chat template proposal).

Reply:
OK, let's reconsider this aspect in the future.

About the new chat template: even if it has its advantages, it still requires internal templatization to some extent, so it would not improve this aspect (differently from what I initially thought).